### PR TITLE
user configurable event buffer size

### PR DESCRIPTION
**Pull request by iknox-fa**
resolves # No ticket-- follow on to #4358 

### Description
Add new global cli flag `EVENT_BUFFER_SIZE` to control the number of events buffered in memory during operation.

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change

### CODE REVIEW

File: core/dbt/events/functions.py [hunk 1]
```diff
@@ -310,13 +309,15 @@ def fire_event(e: Event) -> None:
     # skip logs when `--log-cache-events` is not passed
     if isinstance(e, Cache) and not flags.LOG_CACHE_EVENTS:
         return
+
     # if and only if the event history deque will be completely filled by this event
     # fire warning that old events are now being dropped
     global EVENT_HISTORY
-    if len(EVENT_HISTORY) == ((EVENT_HISTORY.maxlen or 100000) - 1):
+    if len(EVENT_HISTORY) == (flags.EVENT_BUFFER_SIZE - 1):
```
Comment (iknox-fa) [inline core/dbt/events/functions.py:316]: @nathaniel-may Sorry I missed this in the first PR feedback, but this pattern is required or fire_event(EventBufferFull()) will create a recursion loop. :|

Comment (nathaniel-may) [reply core/dbt/events/functions.py:316]: ahh right.

File: test/unit/test_events.py [hunk 2]
```diff
@@ -88,25 +89,29 @@ def test_event_codes(self):
 
 class TestEventBuffer(TestCase):
 
+    def setUp(self) -> None:
+        flags.EVENT_BUFFER_SIZE = 10
```
Comment (nathaniel-may) [inline test/unit/test_events.py:93]: much better

**Review by nathaniel-may — APPROVED** [review]
LGTM

**Comment by iknox-fa** [comment]
@nathaniel-may @jtcohen6 -- Is there anything else additional I need to do to document the new flag?  I'm not sure of the SOP for new ones.

**Comment by nathaniel-may** [comment]
good question. I'm not sure. Maybe @leahwicz knows?

**Comment by leahwicz** [comment]
I'm not sure but let's merge this PR and we can always open another PR for docs

**Comment by emmyoop** [comment]
`main.py` has the help string

**Pull request merged**